crates/youki/src/commands/features.rs

```diff
@@ -1,8 +1,277 @@
+    // List of the recognized hook names, e.g., "createRuntime".
+    hooks: Option<Vec<String>>,
+    // List of the recognized mount options, e.g., "ro".
+    mount_options: Option<Vec<String>>,
+    // Specific to Linux.
+    linux: Option<Linux>,
+    // Contains implementation-specific annotation strings.
+    annotations: Option<std::collections::HashMap<String, String>>,
+}
+
+// Specific to Linux.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Linux {
+    // List of the recognized namespaces, e.g., "mount".
+    namespaces: Option<Vec<String>>,
+    // List of the recognized capabilities, e.g., "CAP_SYS_ADMIN".
+    capabilities: Option<Vec<String>>,
+    cgroup: Option<Cgroup>,
+    seccomp: Option<Seccomp>,
+    apparmor: Option<Apparmor>,
+    selinux: Option<Selinux>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Seccomp {
+    enabled: Option<bool>,
+    actions: Option<Vec<String>>,
+    operators: Option<Vec<String>>,
+    archs: Option<Vec<String>>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Apparmor {
+    enabled: Option<bool>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Selinux {
+    enabled: Option<bool>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Cgroup {
+    v1: Option<bool>,
+    v2: Option<bool>,
+    systemd: Option<bool>,
+    systemd_user: Option<bool>,
+}
+
+// Function to query and return capabilities
+fn query_caps() -> Result<Vec<String>> {
+    let mut available_caps = Vec::new();
+
+    for cap in all() {
+        // Check if the capability is in the permitted set
+        if caps::has_cap(None, CapSet::Permitted, cap).unwrap_or(false) {
+            available_caps.push(format!("{:?}", cap));
+        }
+    }
+
+    Ok(available_caps)
+}
+
+// Function to query and return namespaces
+fn query_supported_namespaces() -> Result<Vec<String>> {
+    let mut supported_namespaces = Vec::new();
+
+    let ns_types = vec!["pid", "net", "uts", "ipc", "mnt", "user", "cgroup", "time"];
+
+    for ns in ns_types {
+        let ns_path = format!("/proc/self/ns/{}", ns);
+        if Path::new(&ns_path).exists() {
+            supported_namespaces.push(ns.to_string());
+        }
+    }
+
+    Ok(supported_namespaces)
+}
 
 /// lists all existing containers
 pub fn features(_: Features) -> Result<()> {
+    // Query supported namespaces
+    let namespaces = match query_supported_namespaces() {
+        Ok(ns) => ns,
+        Err(e) => {
+            eprintln!("Error querying supported namespaces: {}", e);
+            Vec::new()
+        }
+    };
+
+    // Query available capabilities
+    let capabilities = match query_caps() {
+        Ok(caps) => caps,
+        Err(e) => {
+            eprintln!("Error querying available capabilities: {}", e);
+            Vec::new()
+        }
+    };
+
+
+    let features = HardFeatures {
+        oci_version_min: Some(String::from("1.0.0")),
+        oci_version_max: Some(String::from("1.0.2-dev")),
```

Comment:
Why not 1.1.0 ?

Reply:
I hardcoded it when i started so I assume it's progressed since then. I don't have any objection to using 1.1.0 but it seems like runc pulls this from the https://github.com/opencontainers/runtime-spec/blob/main/specs-go/version.go . Should I just hardcode for 1.2.0 (which is the latest) or is there a rust runtime-spec equivalent? 

Thank you for reviewing this!